### PR TITLE
US040 login system

### PR DESCRIPTION
**Pull request by OdhranHague**
UI page created for logging in:
--
Screenshot 2023-12-22 at 14 35 36

Screenshot 2023-12-22 at 14 35 43

Error message whenever username / password is incorrect:

Screenshot 2023-12-22 at 14 36 43

- Whenever login is valid it renders a main menu boiler plate page
- Register button does nothing currently (probably should make a ticket for a registering system)

### CODE REVIEW

File: test/unit/service/authServiceTests.ts
Comment (Joseph-Magee) [inline test/unit/service/authServiceTests.ts:1]: commented out code

File: views/login.html
Comment (Joseph-Magee) [inline views/login.html:1]: remove emplty lines

**Review by Joseph-Magee — APPROVED** [review]
looks good

File: controller/roleController.ts [hunk 1]
```diff
@@ -1,16 +1,31 @@
...
 const roleService = require("../service/roleService");
+const bandService = require("../service/bandService");
 
 module.exports = function(app: Application){
     app.get("/job-roles", async (req: Request, res: Response) => {
         let data: Role[];
+        let dataa: Band[];
         try{
             data = await roleService.getAllRoles();
+            dataa = await bandService.getAllBands();
```
Comment (jkelso662) [inline controller/roleController.ts:14]: typo!

File: controller/roleController.ts [hunk 2]
```diff
@@ -1,16 +1,31 @@
...
         } catch (e) {
             console.log(e);
         }
-        res.render("roles", { roles : data});
+        res.render("roles", { roles : data, bands : dataa});
```
Comment (jkelso662) [inline controller/roleController.ts:18]: typo!

File: test/ui/login.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            } catch (error) {
+                throw new Error("Error in locating or waiting for error message");
+            }
+
+            let errorMessageText = await errorMessage.getText();
+
+            chai.assert.equal(errorMessageText.trim(), "Could not login");
+        } finally {
+            await driver.quit();
+        }
```
Comment (jkelso662) [inline test/ui/login.js:36]: Big function! Consider refactoring at some point but not making it a blocker if you want to carry on

File: service/familyService.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+const familyValidator = require("../validator/familyValidator");
+
+module.exports.createFamily = async function(RoleFamily: RoleFamily): Promise<number> {
+    const error: string = familyValidator.validatefamily(RoleFamily);
+        if (error){
+            throw new Error(error);
+        }
+        
+        try{
+            console.log(RoleFamily);
```
Comment (jkelso662) [inline service/familyService.ts:12]: Should this be in there?

**Review by jkelso662 — CHANGES_REQUESTED** [review]
Really close! Just a couple of changes and this is good to go!

File: public/css/style.css [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+.pad-register-btn {
+    translate: 10px;
+}
+
+.gradient-custom-2 {
+    /* fallback for old browsers */
+    background: #283583;
+
+    /* Chrome 10-25, Safari 5.1-6 */
+    background: -webkit-linear-gradient(to right, 	#283583, #41679f, #009fe3);
```
Comment (jkelso662) [inline public/css/style.css:26]: Random white space here!

**Review by jkelso662 — APPROVED** [review]

**Review by Joseph-Magee — APPROVED** [review]
looks good

**Review by UserConor — COMMENTED** [review]
Great UI! Only small thing is maybe change the copyright to @ 2024 but isn't a requirement

**Pull request closed without merging**